src/jobflow_remote/cli/admin.py

```diff
@@ -30,13 +32,92 @@
     out_console,
 )
 from jobflow_remote.jobs.data import DbCollection
+from jobflow_remote.jobs.upgrade import DatabaseUpgrader
 
 app_admin = JFRTyper(
     name="admin", help="Commands for administering the database", no_args_is_help=True
 )
 app.add_typer(app_admin)
 
 
+@app_admin.command()
+def upgrade(
+    test_version_upgrade: Annotated[
+        Optional[str],
+        typer.Option(
+            help="Explicitly sets the target version for upgrade. For testing purposes only",
+            hidden=True,
+        ),
+    ] = None,
+    no_dry_run: Annotated[
+        bool,
+        typer.Option(
+            "--no-dry-run",
+            "-ndr",
+            help="Skip the dry-run step",
+        ),
+    ] = False,
+    check_env: Annotated[
+        bool,
+        typer.Option(
+            "--check-env",
+            "-ce",
+            help="Report the changes in the package installed in the python environment before "
+            "proceeding with the upgrade",
+        ),
+    ] = False,
+) -> None:
+    """
+    Upgrade the jobflow database.
+    WARNING: can modify all the data. Previous version could not be retrieved anymore.
+    It preferable to perform a backup before upgrading.
+    """
+    check_stopped_runner(error=True)
+
+    jc = get_job_controller()
+    jobflow_check = jc.upgrade_check_jobflow()
+    if jobflow_check:
+        out_console.print(jobflow_check)
+
+    if check_env:
+        full_check = jc.upgrade_full_check()
+        if jobflow_check or full_check:
+            text = Text.from_markup(
+                +full_check
+                + "This does not necessarily pose a problem. Just check if there is any potential "
+                "incompatible upgrade"
+            )
+            out_console.print(text)
+
+    upgrader = DatabaseUpgrader(jc)
+    if not no_dry_run:
+        target_version = test_version_upgrade or upgrader.current_version
+        actions = upgrader.dry_run(target_version=test_version_upgrade)
+        if not actions:
+            out_console.print(
+                f"No actions will be required for upgrading to version {target_version}"
```

Comment:
Indeed it should not happen, but I thought of leaving the "if" just in case things will change in the future. Anyway I can remove it if you think it better.

Reply:
Ok lets keep it